src/test/java/bio/terra/service/filedata/DrsServiceTest.java

```diff
@@ -756,11 +755,14 @@ void lookupObjectByDrsId_bucketNotFoundForGcpSelfHostedSnapshot() {
     verify(samService)
         .verifyAuthorization(
             TEST_USER, IamResourceType.DATASNAPSHOT, snapshotId.toString(), IamAction.READ_DATA);
-    assertThat(
-        exception.getMessage(), startsWith("GCS bucket from %s not found".formatted(sourcePath)));
-    assertThat(
-        exception.getCauses(),
-        containsInAnyOrder(snapshotCacheResult.toString(), googleFsFile.toString()));
```

Comment:
We are confident there will be no secrets in either of these? I think that's the only issue I would potentially see.

Reply:
Good question.  We have a preestablished unit test that verifies that we've checked the caller's read permissions on the underlying snapshot first.

What we've returned here is a subset of information derived from the snapshot summary, which the caller could otherwise obtain via https://data.terra.bio/swagger-ui.html#/snapshots/retrieveSnapshotSummary, and a subset of information derived from the file entry, which the caller could otherwise obtain via https://data.terra.bio/swagger-ui.html#/snapshots/lookupSnapshotFileByPath.